### PR TITLE
host/iso: Add doxygen comments in the header file

### PR DESCRIPTION
**Pull request by wpiet**
Adds missing structures and functions documentation.

### CODE REVIEW

File: nimble/host/include/host/ble_iso.h [hunk 1]
```diff
@@ -19,29 +19,112 @@
...
+     * created. The value shall be between 0x00 and 0xEF.
+     */
     uint8_t big_handle;
+
+    /**
+     * The maximum time in microseconds for transmission of PDUs of all BISes in
+     * a BIG event. The value of BIG_Sync_Delay shall equal the time from the
+     * anchor point to the BIG Synchronization point and shall be between
+     * 0x0000EA and 0x7FFFFF.
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:63]: Ditto, probably just a short description like below would be sufficient 
```suggestion
    /**
     * The maximum time in microseconds for transmission of PDUs of all BISes in
     * a BIG event.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 2]
```diff
@@ -19,29 +19,112 @@
...
     uint32_t big_sync_delay;
+
+    /**
+     * The actual transport latency of transmitting payloads of all BISes in the
+     * BIG in microseconds. The value shall be between 0x0000EA and 0x7FFFFF.
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:69]: Ditto
```suggestion
    /**
     * The actual transport latency of transmitting payloads of all BISes in the
     * BIG in microseconds.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 3]
```diff
@@ -19,29 +19,112 @@
...
     uint32_t transport_latency_big;
+
+    /**
+     * The PHY used to create the BIG. The value shall be one of the following:
+     *     o BLE_HCI_LE_PHY_1M
+     *     o BLE_HCI_LE_PHY_2M
+     *     o BLE_HCI_LE_PHY_CODED
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:77]:
```suggestion
    /**
     * The PHY used to create the BIG. The value can be one of the following:
     *     o BLE_HCI_LE_PHY_1M
     *     o BLE_HCI_LE_PHY_2M
     *     o BLE_HCI_LE_PHY_CODED
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 4]
```diff
@@ -19,29 +19,112 @@
...
     uint8_t phy;
+
+    /**
+     * The number of subevents per BIS in each BIG event. The value shall be
+     * between 1 and 31 and shall be an integer multiple of BN.
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:83]:
```suggestion
    /**
     * The number of subevents per BIS in each BIG event.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 5]
```diff
@@ -19,29 +19,112 @@
...
     uint8_t nse;
+
+    /**
+     * The Burst Number (BN) specifies the number of new payloads in each BIS
+     * event. The value of BN shall be between 1 and 7.
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:89]:
```suggestion

    /**
     * The Burst Number (BN) specifies the number of new payloads in each BIS
     * event.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 6]
```diff
@@ -19,29 +19,112 @@
...
     uint8_t bn;
+
+    /**
+     * The Pre-Transmission Offset (PTO) specifies the offset of groups that
+     * carry data associated with the future BIS events. The value of PTO shall
+     * be between 0 and 15.
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:96]:
```suggestion
    /**
     * The Pre-Transmission Offset (PTO) specifies the offset of groups that
     * carry data associated with the future BIS events.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 7]
```diff
@@ -19,29 +19,112 @@
...
     uint8_t pto;
+
+    /**
+     * The Immediate Repetition Count (IRC) specifies the number of groups that
+     * carry the data associated with the current BIS event. The value of IRC
+     * shall be between 1 and 15.
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:103]:
```suggestion
     * The Immediate Repetition Count (IRC) specifies the number of groups that
     * carry the data associated with the current BIS event.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 8]
```diff
@@ -19,29 +19,112 @@
...
     uint8_t irc;
+
+    /**
+     * The maximum number of data octets (excluding the MIC, if any) that can be
+     * carried in each BIS Data PDU in the BIG. The value shall be between 1 and
+     * 251 octets.
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:110]:
```suggestion
    /**
     * The maximum number of data octets (excluding the MIC, if any) that can be
     * carried in each BIS Data PDU in the BIG.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 9]
```diff
@@ -19,29 +19,112 @@
...
     uint16_t max_pdu;
+
+    /**
+     * The time between two adjacent BIG anchor points in units of 1.25 ms. The
+     * value shall be between 0x0004 and 0x0C80 (i.e. 5 ms to 4 s).
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:116]:
```suggestion
    /**
     * The time between two adjacent BIG anchor points in units of 1.25 ms.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 10]
```diff
@@ -19,29 +19,112 @@
...
     uint16_t iso_interval;
+
+    /**
+     * The total number of BISes in the BIG. The value shall be between 1 and 31.
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:121]:
```suggestion
    /**
     * The total number of BISes in the BIG.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 11]
```diff
@@ -19,29 +19,112 @@
...
     uint8_t num_bis;
+
+    /**
+     * The connection handles of all the BIS in the BIG. The value shall be
+     * between 0x0000 and 0x0EFF.
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:127]:
```suggestion
    /**
     * The connection handles of all the BIS in the BIG.
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 12]
```diff
@@ -83,34 +166,149 @@ struct ble_iso_event {
...
     uint8_t rtn;
+
+    /**
+     * The PHY parameter is a bit field that indicates the PHY used for
+     * transmission of PDUs of BISes in the BIG. The value can be one of the
+     * following:
+     *     o BLE_HCI_LE_PHY_1M_PREF_MASK
+     *     o BLE_HCI_LE_PHY_2M_PREF_MASK
+     *     o BLE_HCI_LE_PHY_CODED_PREF_MASK
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:206]:
```suggestion
    /**
     * The PHY parameter is a bit field that indicates the PHY used for
     * transmission of PDUs of BISes in the BIG. The value shall be one of the
     * following:
     *     o BLE_HCI_LE_PHY_1M_PREF_MASK
     *     o BLE_HCI_LE_PHY_2M_PREF_MASK
     *     o BLE_HCI_LE_PHY_CODED_PREF_MASK
     */
```

File: nimble/host/include/host/ble_iso.h [hunk 13]
```diff
@@ -83,34 +166,149 @@ struct ble_iso_event {
...
     uint8_t phy;
+
+    /**
+     * Indicates the preferred method of arranging subevents of multiple BISes.
+     * The value can be one of the following:
+     *     o 0x00 - Sequential
+     *     o 0x01 - Interleaved
+     */
```
Comment (MariuszSkamra) [inline nimble/host/include/host/ble_iso.h:214]:
```suggestion
    /**
     * Indicates the preferred method of arranging subevents of multiple BISes.
     * The value shall be one of the following:
     *     o 0x00 - Sequential
     *     o 0x01 - Interleaved
     */
```
and in other places

**Review by MariuszSkamra — APPROVED** [review]
LGTM

**Review by KKopyscinski — APPROVED** [review]

**Review by sjanc — APPROVED** [review]

**Pull request merged**